### PR TITLE
Support text alignment 'justify' in BitmapText

### PR DESCRIPTION
**Pull request by shaise**
##### Description of change
Add 'justify' feature to bitmap texts. This aligns the text both right and left using varying spaces between words
roundPixels = true, is recommended for small font sizes. 

##### Pre-Merge Checklist
- [x] Tests and/or benchmarks are included
- [x] Documentation is changed or added - only inside code
- [x] Lint process passed (`npm run lint`)
- [x] Tests passed (`npm run test`)

test_justify
[TestJustify.zip](https://github.com/pixijs/pixi.js/files/5453257/TestJustify.zip)

### CODE REVIEW

**Review by bigtimebuddy — APPROVED** [review]
I'm bummed we don't support justify on Text as well. Hopefully someone can add support on that class too.

**Comment by shaise** [comment]
@bigtimebuddy ,

In fact, I was looking at the standard text as well, to see if I can add it there as well, but it is a bit more complicated as the text is not placed char-by-char, but full lines. Playing with that is more tricky, and I don't want to mess it up.

shai

**Comment by bigtimebuddy** [comment]
No worries, that's a more complicated task, for sure.

File: packages/text-bitmap/src/BitmapText.ts [hunk 1]
```diff
@@ -473,7 +488,7 @@ export class BitmapText extends Container
         for (let i = 0; i < lenChars; i++)
         {
             const char = chars[i];
-            let offset = char.position.x + lineAlignOffsets[char.line];
+            let offset = char.position.x + (lineAlignOffsets[char.line] * char.prevSpaces);
```
Comment (SukantPal) [inline packages/text-bitmap/src/BitmapText.ts:491]: This line is now quite confusing (although I understand in the context of this PR, "adding line justify"). The problem is that `prevSpaces` is 1 for all other alignments, and that is not at all obvious. There may be actually character spaces before `char` on that line, but we are ignoring them for non-justify alignments

I would suggest keeping `prevSpaces` the actual number of spaces before the character. Then, add an ternary operator to multiply it only when the alignment is justify.

```suggestion
            let offset = char.position.x + (lineAlignOffsets[char.line] * (this._align === 'justify' ? char.prevSpaces : 1));
```

File: packages/text-bitmap/src/BitmapText.ts [hunk 2]
```diff
@@ -301,6 +308,7 @@ export class BitmapText extends Container
             charRenderData.charCode = charCode;
             charRenderData.position.x = pos.x + charData.xOffset + (this._letterSpacing / 2);
             charRenderData.position.y = pos.y + charData.yOffset;
+            charRenderData.prevSpaces = this._align === 'justify' ? spaceCount : 1;
```
Comment (SukantPal) [inline packages/text-bitmap/src/BitmapText.ts:311]: Following up on the other suggestion, keep `prevSpaces` consistent across alignements:

```suggestion
            charRenderData.prevSpaces = spaceCount;
```

**Review by SukantPal — CHANGES_REQUESTED** [review]
Thanks @shaise. The changes I suggest will help future readers from loosing context in the code.

File: packages/text-bitmap/src/BitmapText.ts (hunk 2 above)
Comment (shaise) [reply packages/text-bitmap/src/BitmapText.ts:311]: Hi @SukantPal ,
Indeed you are correct. I was just thinking that my way might be a bit more efficient. But on second thought, the added comparison is compensated by one less comparison in your second fix, so I will update the PR with your changes.

**Review by SukantPal — APPROVED** [review]
LGTM!

**Pull request merged**